python/cuml/benchmark/bench_helper_funcs.py

```diff
@@ -45,7 +50,30 @@ def _build_fil_classifier(m, data, arg={}):
 
     # use maximum 1e5 rows to train the model
     train_size = min(data[0].shape[0], 100000)
-    dtrain = xgb.DMatrix(data[0][:train_size, :], label=data[1][:train_size])
+
+    if isinstance(data[0], (pd.DataFrame, pd.Series)):
+        train_data = datagen._convert_to_numpy(data[0])[:train_size, :]
+        train_label = datagen._convert_to_numpy(data[1])[:train_size]
+        dtrain = xgb.DMatrix(train_data, label=train_label)
```

Comment:
This line looks identical everywhere, so feel free to move it to after the conditional.